app/2.3.x/external-plugins.md

```diff
@@ -0,0 +1,247 @@
+---
+title: External plugins support
+---
+
+## Introduction
+
+External plugins are those that run on a process separate from Kong itself,
+enabling the use of any programming language for which an appropriate
+"plugin server" is available.
+
+Each plugin server hosts one or more plugins and communicates with the
+main Kong process via unix sockets.  If so configured, Kong can manage
+those processes, starting, restarting and stopping as necessary.
+
+Kong currently maintains a Go language pluginserver,
+[go-pluginserver] and the corresponding PDK library
+package, [go-pdk].
+
+## Kong configuration
+
+The `pluginserver_names` property is a comma-separated list of names, one
+for each plugin server process.  These names are used to group each process'
+properties and to annotate log entries.
+
+For each name, three other properties can be defined:
+
+property | description | default
+---------|-------------|--------
+`pluginserver_<NAME>_socket` | unix socket path | `/usr/local/kong/<NAME>.socket`
+`pluginserver_<NAME>_start_cmd` | command to start the plugin server process | `/usr/local/bin/<NAME>`
+`pluginserver_<NAME>_query_cmd` | command to dump available plugins' info | `/usr/local/bin/query_<NAME>`
+
+
+For example, you could set Go and Python plugins like this:
+
+```
+pluginserver_names = go,python
+
+pluginserver_go_socket = /usr/local/kong/go_pluginserver.socket
+pluginserver_go_start_cmd = go-pluginserver -kong-prefix /usr/local/kong/ -plugins-dir /usr/local/kong/go-plugins
+pluginserver_go_query_cmd = go-pluginserver -dump-all-plugins -plugins-dir /usr/local/kong/go-plugins
+
+pluginserver_python_socket = /usr/local/kong/pyton_pluginserver.socket
+pluginserver_python_start_cmd = pypluginserver.py
+pluginserver_python_query_cmd = pypluginserver.py -dump
+```
+
+### Legacy configuration
+
+Kong versions 2.0 to 2.2 supported only Go external plugins and a single
+pluginserver using a different configuration style.  Starting with Kong 2.3,
+the old style is recognized and internally transformed to the new style.
+
+If property `pluginserver_names` isn't defined, the legacy properties
+`go_plugins_dir` and `go_pluginserver_exe` are tried:
+
+property | description | default
+---------|-------------|--------
+`go_plugins_dir` | directory with Go plugins | `off`, meaning to disable Go plugins
+`go_pluginserver_exe` | path to the go-pluginserver executable | `/usr/local/bin/go-pluginserver`
+
+Notes:
```

Comment:
Should this be Limitations?

Reply:
not sure.  "limitations" sounds like something we couldn't make better, but this is about backwards compatibility.  i thought of adding a fourth bulletpoint like "kong 2.3 supports go-pluginserver v0.5.0 with the old configuration style and either go-pluginserver v0.6.0 or go-pdk embedded server (or both at the same time)"  but i don't know if that helps to clear or to confuse things.

maybe a table?  in the end it's just: "use the latest of everything, and it should work"

Reply:
Maybe just make the lead-in "Backward compatibility" instead of "Notes:"?

And add your last sentence.